### PR TITLE
docs(Publishing): add automation details and format tweaks

### PR DESCRIPTION
**Pull request by booc0mtaco**
### Summary:

- clarify usage of tags upon release
- mention deprecated packages
- add documentation on github automations
- add additional logic checks on release notes

### CODE REVIEW

File: docs/PUBLISHING.md [hunk 1]
```diff
@@ -27,10 +27,9 @@ We follow a [git-flow](https://nvie.com/posts/a-successful-git-branching-model/)
 
 #### Cutting a release branch
 
-_Warning: In step 4 you will see a message in your terminal with instructions on publising the package. Ignore the message in your terminal and follow the instructions in this guide instead. We send the release branch through the pull request review process before publishing._
-
+0. Before beginning, run `git checkout next && git pull origin next` to ensure you have the most up-to-date changes.
 1. Confirm that all checks are green on CI.
-2. Run `git checkout next && git pull origin next` to ensure you have the most up-to-date changes.
+2. Run `git fetch origin` to retrieve all the latest remote branches and tags.
```
Comment (jinlee93) [inline docs/PUBLISHING.md:32]: Was trying to think of a way to join steps 0 and 2 but this seems safer

File: docs/PUBLISHING.md [hunk 2]
```diff
@@ -42,43 +41,44 @@ yarn release
...
-6. Open a pull request for the release branch, merging into **`main`**. 
+6. Open a pull request for the release branch, merging into **`main`**.
 
-For the commit message, use the new version's content in the [CHANGELOG.md](../CHANGELOG.md) (e.g., all the changes for version 9.0.0).
+For the commit message, use the new version's content in the [CHANGELOG.md](../CHANGELOG.md) (e.g., all the changes for version 9.0.0). Review the content in the changelog to make sure the notes, and the from-version and to-version are correct.
 
 Merge the PR through a **merge commit**:
-   ![github user interface showing a dropdown with the different merge options. the option "create a merge commit" is highlighted](https://user-images.githubusercontent.com/15840841/170514789-4f936ba2-c63d-486c-827a-b9e9e86b612e.png)
+
+![github user interface showing a dropdown with the different merge options. the option "create a merge commit" is highlighted](https://user-images.githubusercontent.com/15840841/170514789-4f936ba2-c63d-486c-827a-b9e9e86b612e.png)
```
Comment (jinlee93) [inline docs/PUBLISHING.md:66]: I think there's some discussion going on regarding the merge strategy / catching up but this is right as of now

**Review by jinlee93 — APPROVED** [review]

File: docs/PUBLISHING.md (hunk 1 above)
Comment (booc0mtaco) [reply docs/PUBLISHING.md:32]: OH, i switched these two around lol. 0 should be to fetch origin, and 2 should remain the same

File: docs/PUBLISHING.md (hunk 2 above)
Comment (booc0mtaco) [reply docs/PUBLISHING.md:66]: yep. this is the same, and just added a new line so that the image has a more legible layout

**Pull request merged**